tasks/files-to-granules/schemas/config.json

```diff
@@ -39,6 +42,10 @@
     "granuleIdExtraction": {
       "type": "string",
       "description": "The regex needed for extracting granuleId from filenames"
+    },
+    "matchFilesWithProducerGranuleId": {
+      "type": "boolean",
+      "description": "Use the 'producerGranuleId' instead of 'granuleId' when mapping files. Defaults to 'false'"
```

Comment:
```suggestion
      "description": "When set to true, use the 'producerGranuleId' instead default behavior of using 'granuleId' when mapping files to granules. Defaults to 'false'"
```

Reply:
Nit: we should also *somewhere* document the potential consequence two incoming config granules having  the same producerGranuleId.

Reply:
Yeah fair. I added a comment here https://bugs.earthdata.nasa.gov/browse/CUMULUS-4079 because we probably need a home for that kind of instruction...